src/System.Runtime.Extensions/tests/System/IO/Path.cs

```diff
@@ -127,12 +198,104 @@ public static void GetTempFileName()
+            Assert.Throws<NotSupportedException>(() => Path.GetFullPath(@"bad:path"));
+
+            // Some Windows-specific bad paths
+            Assert.Throws<ArgumentException>(() => Path.GetFullPath(Path.DirectorySeparatorChar + ".. ." + Path.DirectorySeparatorChar));
+            Assert.Throws<ArgumentException>(() => Path.GetFullPath(Path.DirectorySeparatorChar + ". ." + Path.DirectorySeparatorChar));
+            Assert.Throws<ArgumentException>(() => Path.GetFullPath(Path.DirectorySeparatorChar + " ." + Path.DirectorySeparatorChar));
+            Assert.Throws<ArgumentException>(() => Path.GetFullPath("C:..."));
+            Assert.Throws<ArgumentException>(() => Path.GetFullPath(@"C:...\somedir"));
+            Assert.Throws<ArgumentException>(() => Path.GetFullPath(@"C  :"));
+            Assert.Throws<ArgumentException>(() => Path.GetFullPath(@"C  :\somedir"));
+            Assert.Throws<ArgumentException>(() => Path.GetFullPath(@"bad::$DATA"));
+            Assert.Throws<PathTooLongException>(() => Path.GetFullPath(@"C:\" + new string('a', 255) + @"\"));
+
+            // Some Windows-specific strange but legal paths
+            Assert.Equal(
+                Path.GetFullPath(curDir + Path.DirectorySeparatorChar),
+                Path.GetFullPath(curDir + Path.DirectorySeparatorChar + ". " + Path.DirectorySeparatorChar));
+            Assert.Equal(
+                Path.GetFullPath(Path.GetDirectoryName(curDir) + Path.DirectorySeparatorChar),
+                Path.GetFullPath(curDir + Path.DirectorySeparatorChar + "..." + Path.DirectorySeparatorChar));
+            Assert.Equal(
+                Path.GetFullPath(Path.GetDirectoryName(curDir) + Path.DirectorySeparatorChar),
+                Path.GetFullPath(curDir + Path.DirectorySeparatorChar + ".. " + Path.DirectorySeparatorChar));
+
+            // Windows-specific UNC paths
+            Assert.Equal(@"\\server\share", Path.GetFullPath(@"\\server\share"));
+            Assert.Equal(@"\\server\share", Path.GetFullPath(@" \\server\share"));
+            Assert.Equal(@"\\server\share\dir", Path.GetFullPath(@"\\server\share\dir"));
+            Assert.Equal(@"\\server\share", Path.GetFullPath(@"\\server\share\."));
+            Assert.Equal(@"\\server\share", Path.GetFullPath(@"\\server\share\.."));
+            Assert.Equal(@"\\server\share\", Path.GetFullPath(@"\\server\share\    "));
+            Assert.Equal(@"\\server\  share\", Path.GetFullPath(@"\\server\  share\"));
+            Assert.Throws<ArgumentException>(() => Path.GetFullPath(@"\\"));
+            Assert.Throws<ArgumentException>(() => Path.GetFullPath(@"\\server"));
+            Assert.Throws<ArgumentException>(() => Path.GetFullPath(@"\\server\"));
+            Assert.Throws<ArgumentException>(() => Path.GetFullPath(@"\\server\.."));
+            Assert.Throws<ArgumentException>(() => Path.GetFullPath(@"\\?\GLOBALROOT\"));
+
+            // Windows short paths
```

Comment:
Short path generation may or may not be turned on for the drive running tests. Is there anything we want to do to allow for that here? (I *think* it may be on for "C:" and off for others in recent OS versions- but that is from limited observation and possibly a side effect of my usage of Storage Spaces.)